### PR TITLE
refactor: move typed-message as a package

### PR DESCRIPTION
**Pull request by Jack-Works**
close MASK-705

### CODE REVIEW

File: packages/typed-message/dom/Renderer/utils/TransformContext.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { Transformer, TypedMessage } from '../../../base'
+import { createContext, useContext, useMemo } from 'react'
+import { emptyTransformationContext, TransformationContext } from '../../../base/transformer/context'
```
Comment (UncleBill) [inline packages/typed-message/dom/Renderer/utils/TransformContext.ts]:
```suggestion
import { emptyTransformationContext, TransformationContext } from '../../../base'
```

Leave details behind some folder, and organize our modules in a higher view.

**Comment by Jack-Works** [comment]
@DimensionDev/qa

This is ready for testing.

File: packages/typed-message/dom/Renderer/utils/renderText.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { memo, Fragment, createElement, useContext } from 'react'
+import { RenderFragmentsContext, RenderFragmentsContextType, DefaultRenderFragments } from './RenderFragments'
+import { parseLink } from '../../../base/utils/parseLink'
+import type { TypedMessageAnchor } from '../../../base/extension'
```
Comment (UncleBill) [inline packages/typed-message/dom/Renderer/utils/renderText.tsx]:
```suggestion
import { parseLink, type TypedMessageAnchor } from '../../../base'
```

Comment (Jack-Works) [reply packages/typed-message/dom/Renderer/utils/renderText.tsx]: parseLink is a un-exported utility

**Comment by wenluomask** [comment]
It works but the text style is weird...
Screen Shot 2022-02-23 at 11 14 05 AM

**Comment by Jack-Works** [comment]
@wenluomask hi fixed, please try again

File: packages/mask/src/components/InjectedComponents/DecryptedPostMetadataRender.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    useActivatedPluginsSNSAdaptor.visibility.useNotMinimalMode,
+    (x) => x.DecryptedInspector,
+    MaskPostExtraPluginWrapper,
+)
+export function DecryptedUI_PluginRendererWithSuggestion(props: MetadataRenderProps) {
+    const a = useDisabledPluginSuggestionFromMeta(props.metadata)
+    const b = useDisabledPluginSuggestionFromPost(extractTextFromTypedMessage(props.message), [])
+
+    const suggest = Array.from(new Set(a.concat(b)))
+    console.log(props)
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/DecryptedPostMetadataRender.tsx]:
```suggestion
```

**Comment by guanbinrui** [comment]
Run `npx gulp locale-kit --remove-unused-keys` to solve CI problems.

**Comment by wenluomask** [comment]
Seems front size in original twitter is fixed. No matter how you change the front text, it keeps as same as encrypted context.
Screen Shot 2022-02-23 at 2 43 48 PM

**Comment by Jack-Works** [comment]
> Seems front size in original twitter is fixed. No matter how you change the front text, it keeps as same as encrypted context.

That is based on each Tweet's character count.

**Comment by wenluomask** [comment]
> > Seems front size in original twitter is fixed. No matter how you change the front text, it keeps as same as encrypted context.
> 
> That is based on each Tweet's character count.

Please contact me in Rocket Chat.

**Comment by wenluomask** [comment]
Checked.

**Comment by wenluomask** [comment]
Please check in Jira: https://mask.atlassian.net/browse/MASK-813

**Comment by wenluomask** [comment]
Checked.

**Pull request merged**